master/buildbot/status/web/forms.py

```diff
@@ -34,9 +35,56 @@ def content(self, request, cxt):
     def getChild(self, path, req):
         if path == "forceBuild":
             return ForceBuildDialogPage()
+        elif path == "rebuild":
+            return RebuildDialogPage()
 
+class BuildDialogPage(HtmlResource):
+    def _decodeArgs(self, request, encoding):
+        args = request.args.copy()
+        for name, argl in args.iteritems():
+            if '_branch' in name:
+                args[name] = [ self.decodeFromURL(arg, encoding) for arg in argl ]
+        return args
 
-class ForceBuildDialogPage(HtmlResource):
+    def _getSlaves(self, builder_status):
+        slaves = [s for s in builder_status.getSlaves() if s.isConnected()]
+        slaves = sorted(slaves, key=attrgetter('friendly_name'))
+        return slaves
+
+    def _getBranches(self, args, request):
```

Comment:
I think it some semantic problem. In python we have `args` which symbolize list of arguments and `kwargs` which symbolize keywords arguments - dict. It's my thoughts about it I wonder what do you think about that.

Reply:
I wouldn't call it `args` nor `kwargs`, since those have particular uses in Python.

Either use something like `argDictionary`, or make it work like `kwargs` usually does:

Option 1
```
myArgs = {'branch' : 'bla, 'otherthing': 'blip'}
self._getBranch(argsDict=myArgs, request=request)

def _getBranches(self, argsDict, request):
```

Option 2
```
myArgs = {'branch' : 'bla, 'otherthing': 'blip'}
self._getBranch(request=request, **myArgs)

def _getBranches(self, request, **kwargs):
```

Reply:
So I think these solutions are not that good. 
In option 1 let's not add the type to name of parameter it will create a mess later. In that case I would leave args at it is in that case

In option 2 I love these `kwargs` here, but `getBranches` is simple enough that we should avoid `kwargs` - I mean here we need one obvious element branches from it. 

From the other hand I wonder it's `getBranches` method, rather it should be called getEncodedBranches (in my lame opinion) because it does it. We could even prototype that with `def encodeBranches(self, branches, request)` and in call give it proper list. But to be honest I don't have strong opinion to none of these solutions.